src/frontend/src/views/Upload/components/Metadata/index.tsx

```diff
@@ -33,24 +33,57 @@ import {
   ParseResult,
   SampleIdToWarningMessages,
 } from "./components/ImportFile/parseFile";
+import { analyticsTrackEvent } from "src/common/analytics/methods";
+import {
+  AnalyticsUploadMetadataType,
+  EVENT_TYPES,
+} from "src/common/analytics/eventTypes";
+import { useSelector } from "react-redux";
+import { selectCurrentPathogen } from "src/common/redux/selectors";
 
 export default function Metadata({
   samples,
   namedLocations,
   metadata,
   setMetadata,
+  analyticsFlowUuid,
 }: Props): JSX.Element {
+  const pathogen = useSelector(selectCurrentPathogen);
   const [isValid, setIsValid] = useState(false);
   const [hasImportedMetadataFile, setHasImportedMetadataFile] =
     useState<boolean>(false);
   const [autocorrectWarnings, setAutocorrectWarnings] =
     useState<SampleIdToWarningMessages>(EMPTY_OBJECT);
+  // this is used to track whether the user has manually edited the metadata for analytics
+  const [hasManuallyEditedMetadata, setHasManuallyEditedMetadata] =
+    useState<boolean>(false);
 
   let numberOfDetectedSamples = 0;
   if (samples != null) {
-    numberOfDetectedSamples = Object.keys(samples).length;
+    numberOfDetectedSamples = (Object.keys(samples).length);
   }
 
+  useEffect(() => {
+    const hasMetadataBeenEdited =
+      hasImportedMetadataFile || hasManuallyEditedMetadata;
+    if (!hasMetadataBeenEdited) return;
+
+    type MetadataType = "BOTH" | "MANUAL" | "TSV";
+    let metadataType: MetadataType = "MANUAL";
+    if (hasImportedMetadataFile) metadataType = "TSV";
+    if (hasImportedMetadataFile && hasManuallyEditedMetadata)
+      metadataType = "BOTH";
+    analyticsTrackEvent<AnalyticsUploadMetadataType>(
+      EVENT_TYPES.UPLOAD_METADATA_TYPE,
```

Comment:
should it be the case that if someone imports a metadata file and then manually changes metadata that two events should be fired?

Reply:
hmm good point, i was thinking that it would be ok since this event is linked to the upload flow id, so you could filter out double events easily, but i'll try moving this to the review step to only fire once